test/operator-framework-e2e/operator_framework_test.go

```diff
@@ -0,0 +1,975 @@
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+	When("Create a catalog object and check if the resources are created", func() {
+		It("Create catalog object for the FBC and check if catalog, packages and bundle metadatas are created", func() {
+			bundleVersions := make([]string, len(bundleInfo.bundles))
+			for i, bundle := range bundleInfo.bundles {
+				bundleVersions[i] = bundle.bundleVersion
+			}
+			operatorCatalog, err = createCatalogCheckResources(operatorCatalog, catalogDInfo, bundleVersions)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+	When("Install an operator and check if the operator operations succeed", func() {
+		It("Create an operator object and install it", func() {
+			By("Creating an operator object")
+			operator, err = createOperator(ctx, catalogDInfo.operatorName, operatorAction.installVersion)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking if the operator operations succeeded")
+			checkOperatorOperationsSuccess(operator, catalogDInfo.operatorName, operatorAction.installVersion, bundleInfo.baseFolderPath)
+		})
+	})
+	When("Upgrade an operator to higher version and check if the operator operations succeed", func() {
+		It("Upgrade to a higher version of the operator", func() {
+			By("Upgrading the operator")
+			operator, err = upgradeOperator(ctx, catalogDInfo.operatorName, operatorAction.upgradeVersion)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking if the operator operations succeeded")
+			checkOperatorOperationsSuccess(operator, catalogDInfo.operatorName, operatorAction.upgradeVersion, bundleInfo.baseFolderPath)
+		})
+	})
+	When("Delete an operator", func() {
+		It("Delete an operator", func() {
+			err = deleteOperator(ctx, catalogDInfo.operatorName)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Verifying the operator doesn't exist")
+			Eventually(func(g Gomega) {
+				err = validateOperatorDeletion(catalogDInfo.operatorName)
+				g.Expect(errors.IsNotFound(err)).To(BeTrue())
+			}).Should(Succeed())
+		})
+	})
+	When("Clearing up catalog object and other files formed for the test", func() {
+		It("Clearing up data generated for the test", func() {
+			//Deleting the catalog object and checking if the deletion was successful
+			Eventually(func(g Gomega) {
+				err = deleteAndValidateCatalogDeletion(operatorCatalog)
+				g.Expect(errors.IsNotFound(err)).To(BeTrue())
+			}).Should(Succeed())
+
+			var toDelete []string
+			toDelete = append(toDelete, filepath.Join(catalogDInfo.baseFolderPath, catalogDInfo.catalogDir))               // delete the FBC formed
+			toDelete = append(toDelete, filepath.Join(catalogDInfo.baseFolderPath, catalogDInfo.catalogDir+".Dockerfile")) // delete the catalog Dockerfile generated
+			err = deleteFolderFile(toDelete)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+})
+
+var _ = Describe("Operator Framework E2E for registry+v1 bundles", func() {
```

Comment:
It looks like this is almost identical to `Operator Framework E2E for plain bundles`. Can we have different test setup, but run the same test checks?

Reply:
The bundle structure is different for `registry+v1` and `plain+v0`. Additionally, FBC formation also differs for both as plain bundle doesn't have `opm` support yet. Because of these differences, I guess separate tests for each bundle type would be better. This helps us easily find and track errors(if they occur), as separate tests show each step clearly, instead of combining steps into functions(for running same test checks), which might make things less clear.